core/execution/margin.go

```diff
@@ -98,7 +98,7 @@ func (m *Market) calcMargins(ctx context.Context, pos *positions.MarketPosition,
 }
 
 func (m *Market) updateMargin(ctx context.Context, pos []events.MarketPosition) []events.Risk {
-	price := m.getLastTradedPrice()
+	price := m.getCurrentMarkPrice()
```

Comment:
I don't think this is right. `updateMargin` is, AFAIK, only called from `recheckMargin`, which in turn is the function that updates the margin during the MTM flow, where the last traded price essentially is the de-facto mark price, but during that flow we need both the previous mark price and the new one (ie `getCurrentMarkPrice` and `getLastTradedPrice`)